count_cells_in_matrix.py

```
@@ -122,7 +153,13 @@ def count_cells(matrix_file: str) -> int:
         return None
     barcode_indexes = set(())
     with open(matrix_file, newline='') as csv_file:
-        dialect = csv.Sniffer().sniff(csv_file.read(1024))
+        csv_reader = None
+        try:
+            dialect = csv.Sniffer().sniff(csv_file.read(1024))
+        except csv.Error:
+            pass  # Retry one more time if first read couldn't be sniffed
+        if not csv_reader:
```

This is always true. If I interpret the intend correctly here, it is to sniff on the first 1024 bytes and then sniff the second 1024 bytes if that fails but I don't quite understand how that improves the detection rate. Did we ever witness the case were sniffing the first 1024 bytes failed but sniffing the second 1024 bytes succeeds? If not, why are we coding for a case we haven't actually encountered or haven't provided a test for? I'd rather have the program fail obviously on an unexpected input than fail awkwardly in an untested heuristic.

---

This block was to handle when the first attempt at sniffing the file type failed, which i did experience in my testing however it was with bad test data.
I've reduced this section to try once and fail if an error is encountered.